dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/ConsoleLogHandler.java

```diff
@@ -105,7 +108,16 @@ public void synchronousWrite(Object event) {
 
             //First retrieve a cached JSON  message if possible, if not, format it and store it.
             if (genData.getJsonMessage() == null) {
-                genData.setJsonMessage((String) formatEvent(eventSourceName, CollectorConstants.MEMORY, event, null, MAXFIELDLENGTH));
+                //check if it's in JSON format
+                String json = null;
+                if (event instanceof LogTraceData) {
+                    json = ((LogTraceData) event).getMessage();
+                }
+                if (isJSON(json) && appsWriteJson) {
+                    genData.setJsonMessage(json);
+                } else {
```

Comment:
should start with the cheapest check... which is to see if appsWriteJson is true.  would also be better to put the content of the second if block inside the first if block since it only applies when the first if condition is true.